crates/aptos-crypto/benches/blst.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+#[macro_use]
+extern crate criterion;
+
+use std::cmp::min;
+use std::ops::MulAssign;
+use blst::blst_p1;
+use aptos_crypto::{
+    bls12381,
+    bls12381::ProofOfPossession,
+    test_utils::{random_keypairs, random_subset, KeyPair},
+    traits::{Signature, SigningKey, Uniform},
+    PrivateKey,
+};
+use aptos_crypto_derive::{BCSCryptoHash, CryptoHasher};
+use criterion::{
+    measurement::Measurement, AxisScale, BatchSize, BenchmarkGroup, BenchmarkId, Criterion,
+    PlotConfiguration, Throughput,
+};
+use rand::{distributions, rngs::ThreadRng, thread_rng, Rng};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, CryptoHasher, BCSCryptoHash, Serialize, Deserialize)]
+struct TestAptosCrypto(String);
+
+fn random_bytes(len: usize) -> Vec<u8> {
+    thread_rng().sample_iter(&distributions::Alphanumeric)
+        .take(len)
+        .map(|c|c as u8)
+        .collect()
+}
+
+fn random_p1() -> blst_p1 {
+    let msg = random_bytes(64);
+    let dst = random_bytes(64);
+    let aug = random_bytes(64);
+    let mut point = blst_p1::default();
+    unsafe { blst::blst_hash_to_g1(&mut point, msg.as_ptr(), msg.len(), dst.as_ptr(), dst.len(), aug.as_ptr(), aug.len()); }
```

Comment:
We cannot use `unsafe`... We can avoid this by switching to `blstrs`

Reply:
Let's move the hash-to-group feature into a separate PR!